### PR TITLE
No array value cache

### PR DESCRIPTION
**Pull request by bergos**
- don't cache array values (we cache the array object)
- don't use dist version in test (test works now without build)

**This PR contains also the changes from the [#47](https://github.com/simplerdf/simplerdf/pull/47) PR**

### CODE REVIEW

**Comment by nicola** [comment]
I think we should discuss this.
If I dont specify in the context that this is an array, you should pick the first element instead..

**Comment by bergos** [comment]
If we don't treat these values as array, the other values will be lost on update.

Automatically using an array is more JSON-LD like.

File: test/test.js [hunk 1]
```diff
@@ -129,6 +129,33 @@ describe('simplerdf', () => {
...
+    graph.add(rdf.createTriple(
+      rdf.createNamedNode(blogIri),
+      rdf.createNamedNode('http://schema.org/name'),
+      rdf.createLiteral('simple blog second name')))
+
+    let blog = simple(blogContext, blogIri, graph)
+
+    let names = blog.name
+
+    assert(simple.isArray(names))
```
Comment (nicola) [inline test/test.js:144]: `name` is defined as `name: 'http://schema.org/name'`.
This means that this is not an array.. :S

Comment (bergos) [inline test/test.js:144]: Yes, but that test works as intended. We will always have a drawback, if the predicate is not defined as an array, but there are two or more triples with that predicate in an existing graph.

I see two solutions:
1. Treat that predicate like there is only one triple, which could lead into data loss.
2. Return a unexpected data type, which must be handled in the code.

So what's the smaller problem? I vote for 2. and also JSON-LD does it that way.

The best solution would be validating the data when it's generated. But sometimes you can't control that, but than you should be prepared for "strange" data.

**Comment by bergos** [comment]
I created a new [PR](https://github.com/simplerdf/simplerdf/pull/50) that contains only the array cache stuff.

**Pull request closed without merging**